### PR TITLE
Add migration test 3.3 -> 3.4

### PR DESCRIPTION
**Pull request by burqen**

### CODE REVIEW

File: community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java [hunk 1]
```diff
@@ -155,6 +239,17 @@ private void verifyIndexes( GraphDatabaseAPI db, Label label ) throws Exception
         assertEquals( 34, countIndexedNodes( db, label, KEY1, KEY2 ) );
     }
 
+    private void verifyAfterAdditionalUpdate( GraphDatabaseAPI db, Label label ) throws Exception
+    {
+        // There should be an index for the label and KEY1 containing 100 nodes
```
Comment (tinwelint) [inline community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java]: Comment is wrong, I realize that it's twice those numbers now that one more round of updates have been added. Comment could reflect that, or be removed.

File: community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java [hunk 2]
```diff
@@ -155,6 +239,17 @@ private void verifyIndexes( GraphDatabaseAPI db, Label label ) throws Exception
...
+        assertTrue( hasIndex( db, label, KEY1 ) );
+        assertEquals( 200, countIndexedNodes( db, label, KEY1 ) );
+
+        // There should be an index for the label and KEY1+KEY2 containing 34 nodes
```
Comment (tinwelint) [inline community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java]: Likewise here

**Review by tinwelint — COMMENTED** [review]
Should the test also assert that the old index is still for the old index provider?

**Comment by burqen** [comment]
@tinwelint Comments addressed and rebase complete

**Pull request merged**